### PR TITLE
Promote first GGUF MoE architecture cohort

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- promote float GGUF graph routes for `arctic`, `dbrx`, `ernie4_5-moe`, and `nomic-bert-moe`
- preserve architecture-specific routing order, normalization, expert layouts, activations, norms, RoPE, shared-expert behavior, and layer schedules
- add strict config/tensor/shape closure and CPU graph execution coverage
- keep preserved quantized import rejected and runtime support deferred for all four routes
- retain normal Hugging Face revision behavior: no production model-ID revision pins, defaults, or rejection gates

## Evidence

- Arctic: pinned llama.cpp synthetic architecture fixture
- DBRX: pinned 64 MiB bounded header from `dbrx-instruct.i1-IQ1_S.gguf`
- ERNIE 4.5 MoE: pinned full `ERNIE-4.5-21B-A3B-Thinking.i1-IQ1_S.gguf`
- NomicBERT-MoE: pinned full `nomic-embed-text-v2-moe.Q2_K.gguf`
- total transferred payload: 4,816,875,008 bytes, below the 16 GiB cap; payloads were removed after evidence collection

## Validation

- broad serial suite: `8662 passed, 64 skipped, 12 deselected`
- focused cohort, attention, registry, census, Kimi K3 compatibility, and CPU execution tests pass
- `lintrunner f --output oneline --all-files`
- generated GGUF documentation check passes
- changed-source mypy was run; the repository remains baseline-red in pre-existing diagnostics outside the cohort-added source lines
- two independent reviews completed; ERNIE ignored output-bias handling and Arctic stacked-expert loading findings were fixed and re-reviewed

### CODE REVIEW

File: src/mobius/models/moe.py [hunk 1]
```diff
@@ -209,13 +213,17 @@ class MoETextModel(nn.Module):
...
+        norm_class: Callable[..., nn.Module] = RMSNorm,
+        layer_class: Callable[..., nn.Module] | None = None,
     ):
         super().__init__()
         self._dtype = config.dtype
         self.embed_tokens = embedding_for_config(config)
+        num_local_experts = config.num_local_experts
+        num_experts_per_tok = config.num_experts_per_tok
+        if num_local_experts is None or num_experts_per_tok is None:
+            raise ValueError("MoE decoder requires expert_count and expert_used_count")
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/moe.py]: Updated the diagnostic to name `num_local_experts` and `num_experts_per_tok`, matching the validated config fields for GGUF and non-GGUF callers.

File: src/mobius/integrations/gguf/_builder.py [hunk 2]
```diff
@@ -2692,6 +2693,507 @@ def _raise_for_invalid_conventional_moe_tensor_contract(gguf_model) -> None:
...
+        layers = int(metadata[f"{architecture}.block_count"])
+        heads = int(metadata[f"{architecture}.attention.head_count"])
+        kv_heads = int(metadata.get(f"{architecture}.attention.head_count_kv", heads))
+        experts = int(metadata[f"{architecture}.expert_count"])
+        top_k = int(metadata[f"{architecture}.expert_used_count"])
+        frequency = int(metadata[f"{architecture}.moe_every_n_layers"])
+        token_types = int(metadata.get("tokenizer.ggml.token_type_count", 0))
+        vocab = int(metadata.get(f"{architecture}.vocab_size", 0)) or len(
+            metadata.get("tokenizer.ggml.tokens", ())
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_builder.py:3012]: Added an explicit positive `tokenizer.ggml.token_type_count` check and a regression test for missing metadata, so this now fails with an actionable field-specific error.

File: src/mobius/models/gguf_encoders.py [hunk 3]
```diff
@@ -383,6 +478,84 @@ def __init__(self, config: ArchitectureConfig):
...
+    def __init__(self, config: ArchitectureConfig, *, routed: bool):
+        super().__init__()
+        self.attention = _FusedPostNormAttention(config)
+        self.attention_output_norm = LayerNorm(config.hidden_size, eps=config.rms_norm_eps)
+        self.mlp: nn.Module
+        if routed:
+            num_local_experts = config.num_local_experts
+            num_experts_per_tok = config.num_experts_per_tok
+            if num_local_experts is None or num_experts_per_tok is None:
+                raise ValueError("NomicBERT-MoE requires expert_count and expert_used_count")
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/gguf_encoders.py]: Updated the NomicBERT-MoE diagnostic to name `num_local_experts` and `num_experts_per_tok` directly.
